### ISSUE TITLE
VRP with Time Windows - Different start and end locations

### ISSUE THREAD

**Issue by ilaif** [google/or-tools#458]
Hey guys,

How do I define an "end" location - meaning a location that the algorithm must finish at (in addition to start location).

Thanks,
Ilai.

**Comment by orwant** [comment]
https://developers.google.com/optimization/routing/tsp/vehicle_routing shows some example code that can be modified to define an end location.

**Comment by ilaif** [comment]
Hey, @orwant - I've gone over this article already but it doesn't explicitly show how. Only how to relax constraints to make it possible to start and end in any location. I would like to restrict the one location to start location and another to end of the route.
If I'm thinking about simple graph problems, I guess that I can just define another dummy location called end, that will have 0 distance to the desired end location, and MAX_DOUBLE distance to all other locations.
I guess that will work. Any thoughts or caveats?

**Comment by orwant** [comment]
You could do that, but shouldn't have to. You can specify end nodes in the RoutingModel constructor:
https://developers.google.com/optimization/reference/constraint_solver/routing/RoutingModel/

From that page:

RoutingModel(int nodes, int vehicles, const std::vector<NodeIndex>& starts, const std::vector<NodeIndex>& ends)

Constructor taking vectors of start nodes and end nodes for each vehicle route. Used to model multiple depots.

**Comment by gsavvid** [comment]
Hello,

Thank you very much for this great library. I'm having an issue as well when trying to solve VRP with time windows and custom start and end locations.

Some background:

1. I'm using the Java version of the library and I am basically translating the Python examples posted [here](https://developers.google.com/optimization/routing/tsp/routing_tasks) and [here](https://developers.google.com/optimization/routing/tsp/vehicle_routing_time_windows).
2. I can successfully solve the VRP with:
2.1 Common start/end locations (depot).
2.2 Arbitrary start/end locations.
2.3 Custom start/end locations.
2.4 Common start/end locations with time windows.
2.5 Arbitrary start/end locations with time windows.

My problem like I mentioned above is when trying to implement custom start/end locations WITH time windows. I'm using the values that the examples are also using but I get the following exception:
```
Exception in thread "main" com.google.ortools.constraintsolver.Solver$FailException: fail
	at com.google.ortools.constraintsolver.operations_research_constraint_solverJNI.IntExpr_setRange(Native Method)
	at com.google.ortools.constraintsolver.IntExpr.setRange(IntExpr.java:59)
```
In other words, when it comes to time windows the problem occurs as soon as I change the `RoutingModel` constructor from `RoutingModel(int nodes, int vehicles, NodeIndex depot)` to `RoutingModel(int nodes, int vehicles, const std::vector<NodeIndex>& starts, const std::vector<NodeIndex>& ends)`.

Any suggestions would be greatly appreciated.
